File: app/src/main/java/com/letianpai/robot/wificonnet/adapter/CustomAdapter.java
package com.letianpai.robot.wificonnet.adapter;

import android.text.TextUtils;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.letianpai.robot.wificonnet.R;
import com.letianpai.robot.wificonnet.callback.KeyPressCallback;
import com.letianpai.robot.wificonnet.view.KeyButton;

import java.util.List;
/**
 * @author liujunbin
 */
public class CustomAdapter extends RecyclerView.Adapter<CustomAdapter.ViewHolder> {

    private List<String> data;

    public CustomAdapter(List<String> data) {
        this.data = data;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
//        holder.textView.setText(data.get(position));
        holder.keyButton.setText(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        KeyButton keyButton;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            keyButton = itemView.findViewById(R.id.keyButton);
            keyButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    String buttonText = ((Button) v).getText() + "";
                    if (!TextUtils.isEmpty(buttonText)){
                        KeyPressCallback.getInstance().pressKey(KeyPressCallback.KEY_TYPE_VALUE,buttonText);
                    }
//                    Log.e("letianpai_test","buttonText: "+ buttonText);
                }
            });

        }
    }
}

File: app/src/main/java/com/letianpai/robot/wificonnet/adapter/CustomAdapter.kt
package com.letianpai.robot.wificonnet.adapter

import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.recyclerview.widget.RecyclerView
import com.letianpai.robot.wificonnet.R
import com.letianpai.robot.wificonnet.callback.KeyPressCallback
import com.letianpai.robot.wificonnet.view.KeyButton

/**
 * @author liujunbin
 */
class CustomAdapter(private val data: List<String>) :
    RecyclerView.Adapter<CustomAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
//        holder.textView.setText(data.get(position));
        holder.keyButton.text = data[position]
    }

    override fun getItemCount(): Int {
        return data.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var keyButton: KeyButton = itemView.findViewById(R.id.keyButton)

        init {
            keyButton.setOnClickListener { v ->
                val buttonText = (v as Button).text.toString() + ""
                if (!TextUtils.isEmpty(buttonText)) {
                    KeyPressCallback.instance
                        .pressKey(KeyPressCallback.KEY_TYPE_VALUE, buttonText)
                }
                //                    Log.e("letianpai_test","buttonText: "+ buttonText);
            }
        }
    }
}
